### PR TITLE
divelogexportlogic: show warning on missing 'theme' folder

### PR DESCRIPTION
**Pull request by neolit123**
### Describe the pull request:

- [x] Bug fix
- [ ] Functional change
- [ ] New feature
- [ ] Code cleanup
- [ ] Build system change
- [ ] Documentation change
- [ ] Language translation

### Pull request long description:

show a warning if the user is trying to export HTML, but the "theme" folder is missing for some reason.
not really user facing, unless someone messes with his installation.

### CODE REVIEW

File: core/divelogexportlogic.cpp [hunk 1]
```diff
@@ -148,8 +148,10 @@ void exportHtmlInitLogic(const QString &filename, struct htmlExportSetting &hes)
 	export_HTML(qPrintable(json_dive_data), qPrintable(photosDirectory), hes.selectedOnly, hes.listOnly);
 
 	QString searchPath = getSubsurfaceDataPath("theme");
-	if (searchPath.isEmpty())
+	if (searchPath.isEmpty()) {
+		report_error(gettextFromC::tr("Cannot find a folder called 'theme' in the standard locations").toUtf8());
```
Comment (dirkhh) [inline core/divelogexportlogic.cpp]: I thought ::tr always returns utf8... so I'm not sure why you need the toUtf8() call...

File: core/divelogexportlogic.cpp [hunk 2]
```diff
@@ -148,8 +148,10 @@ void exportHtmlInitLogic(const QString &filename, struct htmlExportSetting &hes)
 	export_HTML(qPrintable(json_dive_data), qPrintable(photosDirectory), hes.selectedOnly, hes.listOnly);
 
 	QString searchPath = getSubsurfaceDataPath("theme");
-	if (searchPath.isEmpty())
+	if (searchPath.isEmpty()) {
+		report_error(qPrintable(gettextFromC::tr("Cannot find a folder called 'theme' in the standard locations")));
```
Comment (neolit123) [inline core/divelogexportlogic.cpp]: switched to `qPrintable`.
`.toUtf8()` converted to `QByteArray` which was then auto-casted to `const char*` which is what `report_error()` needed.

File: core/divelogexportlogic.cpp (hunk 1 above)
Comment (neolit123) [reply core/divelogexportlogic.cpp]: switched to `qPrintable`.
`.toUtf8()` converted to `QByteArray` which was then auto-casted to `const char*` which is what `report_error()` needs.

**Comment by dirkhh** [comment]
that looks better... I guess we should let Travis run, though I don't see what could be wrong here.

File: core/divelogexportlogic.cpp (hunk 2 above)
Comment (bstoeger) [inline core/divelogexportlogic.cpp:152]: If it is not user-facing, is it worth adding a translation string?

Comment (neolit123) [reply core/divelogexportlogic.cpp:152]: it can be.

Comment (bstoeger) [reply core/divelogexportlogic.cpp:152]: If you think a new translation string  is warranted, I think we can commit this (after all @dirkhh gave his approval).

**Pull request merged**